### PR TITLE
choose n_jobs for ensemble according to n_jobs per learner

### PR DESCRIPTION
**Pull request by sonichi**
To prevent the test error in https://github.com/microsoft/FLAML/runs/6425372138?check_suite_focus=true#step:8:165
Catch the ensemble error and log the error.
Related to #476

### CODE REVIEW

**Review by Qiaochu-Song — APPROVED** [review]

File: flaml/automl.py [hunk 1]
```diff
@@ -534,10 +534,11 @@ def custom_metric(
...
                  search.
-             ensemble: boolean or dict | default=False. Whether to perform
-                 ensemble after search. Can be a dict with keys 'passthrough'
-                 and 'final_estimator' to specify the passthrough and
-                 final_estimator in the stacker.
+            ensemble: boolean or dict | default=False. Whether to perform
+                ensemble after search. Can be a dict with keys 'passthrough'
+                and 'final_estimator' to specify the passthrough and
+                final_estimator in the stacker. The dict can also contain
+                'n_jobs' as the key to specify the n_jobs for the stacker.
```
Comment (liususan091219) [inline flaml/automl.py:541]:
```suggestion
                'n_jobs' as the key to specify the number of jobs for the stacker.
```

File: flaml/automl.py [hunk 2]
```diff
@@ -3174,18 +3183,34 @@ def _search(self):
...
+                    n_cpus = (
+                        ray.is_initialized()
+                        and ray.available_resources()["CPU"]
+                        or os.cpu_count()
+                    )
+                else:
+                    n_cpus = os.cpu_count()
+                n_jobs = (
+                    -self._state.n_jobs
+                    if abs(self._state.n_jobs) == 1
```
Comment (liususan091219) [inline flaml/automl.py:3198]: why n_jobs = -self._state.n_jobs? This is unclear, can you add comments?

Comment (liususan091219) [reply flaml/automl.py:3198]: so when n_jobs is max, minimize it; maximize it when min? why?

Comment (liususan091219) [reply flaml/automl.py:3198]: Unresolved.

Comment (sonichi) [reply flaml/automl.py]: What about now?

**Pull request merged**